src/backend/executor/nodeDML.c

```diff
@@ -281,17 +281,29 @@ ExecInitDML(DML *node, EState *estate, int eflags)
 			dmlstate->cleanedUpSlot);
 
 	/*
-	 * We don't maintain typmod in the targetlist, so we should fixup the
-	 * junkfilter to use the same tuple descriptor as the result relation.
-	 * Otherwise the mismatch of tuple descriptor will cause a break in
-	 * ExecInsert()->reconstructMatchingTupleSlot().
+	 * The comment below is related to ExecInsert() and ExecUpdate(). The code
+	 * works correctly, because insert and update operations always translate
+	 * full set of attrs to targetlist. So, resulting tupledesc has the same
+	 * number of attrs after replacing. ExecDelete() doesn't have such
+	 * machinery, and more, can work with subset of table attrs. So, the code
+	 * below can add attrs, which doesn't exist in original targetlist. This
+	 * may cause execution errors.
```

Comment:
I propose to rewrite the comment (e.x ExecUpdate() is not yet executed for the ORCA generated):

The code below is related only to INSERT/UPDATE operations, since they use the full set of relation attributes during the execution.  DELETE operation doesn't require all the attributes, what is represented by a shorter targetList that contains only junk entries.

Reply:
But my idea was to link this comment with another comment, which explains the code below by itself. I chose this to not rewrite original comment. So, I use ExecInsert() and ExecUpdate() (not INSERT/UPDATE), because this was already used in linked comment.

I think we don't need to mention triggers here as it doesn't matter is there any trigger or not, code works the same.